src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -594,10 +594,53 @@ public unsafe void Reset()
         {
             _updateStopwatch.Stop();
             _renderStopwatch.Stop();
-            _glfw.DestroyWindow(_windowPtr);
+
+            try
+            {
+                _glfw.DestroyWindow(_windowPtr);
+            }
+#pragma warning disable 168
+            // If the window is already destroyed, it throws an exception,
+            // but we want the window destroyed anyways, so just ignore it
+            catch(GlfwException _e)
+#pragma warning restore 168
+            {
+                
+            }
+            
             _windowPtr = (WindowHandle*) 0;
         }
 
+        private void Dispose(bool disposing)
+        {
+            unsafe
+            {
+                Reset();
+                _glfw.GcUtility.Unpin(_onClosing);
+                _glfw.GcUtility.Unpin(_onMaximized);
+                _glfw.GcUtility.Unpin(_onMinimized);
+                _glfw.GcUtility.Unpin(_onMove);
+                _glfw.GcUtility.Unpin(_onResize);
+                _glfw.GcUtility.Unpin(_onFileDrop);
+                _glfw.GcUtility.Unpin(_onFocusChanged);
+            }
+            if (disposing)
+            {
+                _glfw?.Dispose();

```

Comment:
Isn't `_glfw` a reference to `GlfwProvider.GLFW.Value`? Wouldn't this break all other windows when disposing one?

Reply:
Rider did that automatically when implementing the dispose pattern, oops.